arho_feature_template/core/plan_manager.py

```diff
@@ -416,21 +486,27 @@ def save_regulation_group(regulation_group: RegulationGroup, plan_id: str | None
             regulation.regulation_group_id_ = feature["id"]  # Updating regulation group ID
             save_regulation(regulation)
 
+    # Handle propositions
+    if regulation_group.propositions:
+        for proposition in regulation_group.propositions:
+            proposition.regulation_group_id_ = feature["id"]  # Updating regulation group ID
+            save_proposition(proposition)
+
     return feature
 
 
 def save_regulation_group_as_config(regulation_group: RegulationGroup):
     pass
 
 
-def save_regulation_group_association(regulation_group_id: str, layer_name: str, feature_id: str) -> QgsFeature:
+def save_regulation_group_association(regulation_group_id: str, layer_name: str, feature_id: str):
     feature = RegulationGroupAssociationLayer.feature_from(regulation_group_id, layer_name, feature_id)
+    if not feature:
+        return
```

Comment:
Voisiko tarkistuksen onko assosiaatio jo olemassa tehdä tässä? Mielestäni olisi selkeämpi jotenkin näin:
```python
def save_regulation_group_association(regulation_group_id: str, layer_name: str, feature_id: str):
    if association_exists(regulation_group_id, layer_name, feature_id):
        break
    feature = RegulationGroupAssociationLayer.feature_from(regulation_group_id, layer_name, feature_id)
    _save_feature(feature, ...)
```